src/test/java/com/skalashynski/spring/springboot/service/StudentServiceTest.java

```diff
@@ -0,0 +1,231 @@
+package com.skalashynski.spring.springboot.service;
+
+import com.skalashynski.spring.springboot.entity.Student;
+import com.skalashynski.spring.springboot.exception.ApiException;
+import com.skalashynski.spring.springboot.repository.StudentRepository;
+import com.skalashynski.spring.springboot.service.impl.StudentServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.BDDMockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class StudentServiceTest {
+
+    @Mock
+    private StudentRepository studentRepository;
+
+    @InjectMocks
+    private StudentServiceImpl studentService;
+
+    private static Stream<Student> provideStudents() {
+        return Stream.of(
+            new Student(Long.MIN_VALUE, "firstName", "lastName",
+                        LocalDate.of(1993, 1, 17), LocalDateTime.now()),
+            new Student(Long.MAX_VALUE, "someName", "lastName",
```

Comment:
create a const for `LocalDateTime.now()` and use it. Don't produce many different objects.

Reply:
done